addons/web_editor/controllers/main.py

```diff
@@ -567,6 +567,16 @@ def _load_resources(self, file_type, views, bundles_restriction, only_user_custo
 
         return files_data_by_bundle
 
+    @http.route('/web_editor/fetch_image', type="json", auth="user", website=True)
+    def fetch_image(self, url):
+        try:
+            req = requests.get(url, timeout=2.5)
+            req.raise_for_status()
+        except requests.exceptions.RequestException as error:
+            logger.warning("Could not get the image data at url %s: %s", url, error)
+            return {'mimetype': '', 'base64Data': ''}
+        return {'mimetype': req.headers['Content-Type'], 'base64Data': b64encode(req.content)}
```

Comment:
@odoo/rd-security Still in development and still to be reviewed but I could use your opinion on the idea here. At minimum I guess a check to see if the user is a restricted user is required here but is this a good idea? The goal is for the client-side editor to fetch images that are CORS protected.

Reply:
Hello, I am not fan of giving free SSRF to anyone, this could lead to security issues, I'll avoid to develop more in public.

Is this feature really needed? And if so, why not limiting it more for example like the method `load_remote_url` in the abstract model `ir.qweb.field.image`.
You can see that the downloaded image is parsed with PIL to make sure it is an image and it will never return the result if not. (i.e. load a sensitive webpage reachable only on local network and get its result will be mitigated here, but still the page has been called)

Reply:
And do we have a good reason to bypass CORS protected pages? I don’t like the idea that the image goes through our server for no reason. If there is a policy reason, we can ask the user to download it and to upload it manually.

Reply:
> And do we have a good reason to bypass CORS protected pages?

@mart-e The good reason is the UX, asking the user to download it and to upload it manually is... meh. And many users won't even understand what we are asking, that's a guarantee. We currently actually don't do that: there is a whole system of creating the attachment on our side with the reference to the original (cross-protected) URL and serving it from the original server through a redirect when using this internal image. That also prevents us to properly make those image options work (that system was there before cross-protection concerns and used for everything and it now only serves the cross-protection purpose). That's our reason to try and get rid of that system.

Now, of course, the question is the security(?) / ethic(?) of it. TBH I don't understand what CORS on images is achieving in the case where any proxy server can access it anyway. In the case our server can access the image, what would be the reason to not do this?

> FWIW you can do that easily enough (and more efficiently in some ways) using GET: if you set stream=True [...]

That's interesting :+1: